apps/omg_watcher/lib/omg_watcher/utxo_exit/core.ex

```diff
@@ -0,0 +1,96 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Watcher.UtxoExit.Core do
+  @moduledoc """
+  Module provides API for compose exit
+  """
+
+  alias OMG.Block
+  alias OMG.State.Transaction
+  alias OMG.Utxo
+  alias OMG.Watcher.DB
+  use OMG.Utils.Metrics
+  require Utxo
+
+  @decorate measure_event()
+  @spec compose_output_exit(list(%DB.Transaction{}) | list(Transaction.Signed.tx_bytes()), Utxo.Position.t()) ::
+          {:error, :utxo_not_found}
+          | {:ok,
+             %{
+               utxo_pos: non_neg_integer(),
+               txbytes: binary,
+               proof: binary,
+               sigs: binary
+             }}
+  def compose_output_exit([%{txindex: _} | _] = txs, Utxo.position(_, txindex, _) = utxo_pos) do
+    if Enum.any?(txs, &match?(%{txindex: ^txindex}, &1)) do
+      sorted_tx_bytes = Enum.sort_by(txs, & &1.txindex) |> Enum.map(& &1.txbytes)
+      compose_output_exit(sorted_tx_bytes, utxo_pos)
+    else
+      {:error, :utxo_not_found}
+    end
+  end
+
+  @decorate measure_event()
+  def compose_output_exit(sorted_tx_bytes, Utxo.position(_blknum, txindex, _) = utxo_pos) do
+    if signed_tx = Enum.at(sorted_tx_bytes, txindex) do
```

Comment:
a proper use of pattern matching is using a `case` clause with `^` operator.

Reply:
here I don't use pattern matching, then I don't need to use `^`.

Reply:
what is it then?

Reply:
check if is different from nil ant give name to use it lower

Reply:
that's kinda confusing that you're making assignments in an if clause, no?

Reply:
That's why i changed it.
Although I'm a bit surprised because assignments in an if clause is very typical.
even some of these languages ​​add such a possibility to reduce the scope of the variable

Reply:
I'm opting to leave the `if new_var_when_not_nil = smth()` kind of code pattern we already using elsewhere, and you can see in elixir/python code.

Reply:
I've definitely never seen it before and it seems counter intuitive.

Reply:
for me is quite intuitive :smile: 
in elixir-lang often occurs (170 matches in 83 files)

Reply:
Cool! I'll adopt it then!